lib/rules/number-no-trailing-zeros/index.js

```diff
@@ -50,28 +52,69 @@ const rule = function(actual) {
           return;
         }
 
-        const match = /(\.\d*)0+(?:\D|$)/.exec(valueNode.value);
-
+        const match = /\.(\d*?)(0+)(?:\D|$)/.exec(valueNode.value);
+        // match[1] is any numbers between the decimal and our trailing zero, could be empty
+        // match[2] is our trailing zero(s)
         if (match === null) {
           return;
         }
 
-        report({
-          message: messages.rejected,
-          node,
-          index:
-            getIndex(node) +
-            valueNode.sourceIndex +
-            match.index +
-            match[1].length,
-          result,
-          ruleName
-        });
+        // our index is:
+        //  the index of our valueNode +
+        //  the index of our match +
+        //  1 for our decimal +
+        //  the length of our potential non-zero number match (match[1])
+        const index = valueNode.sourceIndex + match.index + 1 + match[1].length;
+
+        // our startIndex is identical to our index except when we have only
+        // trailing zeros after our decimal. in that case we don't need the decimal
+        // either so we move our index back by 1.
+        const startIndex = match[1].length > 0 ? index : index - 1;
+
+        // our end index is our original index + the length of our trailing zeros
+        const endIndex = index + match[2].length;
+
+        if (context.fix) {
+          fixPositions.unshift({
+            startIndex,
+            endIndex
+          });
+          return;
+        } else {
+          report({
+            message: messages.rejected,
+            node,
+            // this is the index of the _first_ trailing zero
+            index: getIndex(node) + index,
+            result,
+            ruleName
+          });
+        }
       });
+
+      if (fixPositions.length) {
+        fixPositions.forEach(function(fixPosition) {
+          const startIndex = fixPosition.startIndex;
+          const endIndex = fixPosition.endIndex;
+          if (node.type === "atrule") {
+            node.params = removeTrailingZeros(
+              node.params,
+              startIndex,
+              endIndex
+            );
+          } else {
+            node.value = removeTrailingZeros(node.value, startIndex, endIndex);
+          }
+        });
+      }
     }
   };
 };
 
+function removeTrailingZeros(input, startIndex, endIndex) {
```

Comment:
Maybe best place `utils`?

Reply:
Is it used by more than rule? If so, then yes I agree. Otherwise, it's probably more helpful to maintainers to keep it here.